### PR TITLE
fix: build devnet amplifier

### PR DESCRIPTION
**Pull request by npty**
# Description

This PR fixes all build errors we accumulate during sui integration. All build errors after this PR should be minimal and simple to fix.

### CODE REVIEW

File: apps/maestro/src/features/InterchainTokenDeployment/steps/deploy-and-register/DeployAndRegister.tsx [hunk 1]
```diff
@@ -105,14 +107,16 @@ export const Step2: FC = () => {
...
-        if (result?.digest) {
+        if (result) {
           rootActions.setTxState({
-            status: "submitted",
+            type: "deployed",
             suiTx: result,
             txHash: result.digest,
-            chainId: sourceChain.chain_id,
+            // TODO: get token address
+            tokenAddress: "0x",
```
Comment (npty) [inline apps/maestro/src/features/InterchainTokenDeployment/steps/deploy-and-register/DeployAndRegister.tsx:118]: @SGiaccobasso could you pls help me set the token address correctly here?

File: apps/maestro/src/features/RegisterRemoteTokens/RegisterRemoteTokens.tsx [hunk 2]
```diff
@@ -94,33 +100,35 @@ export const RegisterRemoteTokens: FC<RegisterRemoteTokensProps> = (props) => {
     await recordRemoteTokenDeployment({
       tokenAddress: props.tokenAddress,
       chainId: props.originChainId ?? -1,
-      deploymentMessageId: `${digest}-${txIndex}`,
+      // TODO: find event Txindex correctly
+      deploymentMessageId: `${digest}`,
```
Comment (npty) [inline apps/maestro/src/features/RegisterRemoteTokens/RegisterRemoteTokens.tsx:104]: the event index needed to be appended here as well

Comment (SGiaccobasso) [reply apps/maestro/src/features/RegisterRemoteTokens/RegisterRemoteTokens.tsx:104]: there's is no transaction index in sui, should we hardcode a 0?

Comment (npty) [reply apps/maestro/src/features/RegisterRemoteTokens/RegisterRemoteTokens.tsx:104]: there's event index. for example, this tx https://devnet-amplifier.axelarscan.io/gmp/AruX9g1GWvSQmF1ZzZuW1uaTEi3fiJXgkWMEFXmKF4vU-1, In this url, you can notice that the format is `digest-eventIndex`. The eventIndex `1` is from the order of `ContractCall` event in this [sent tx](https://suiscan.xyz/testnet/tx/AruX9g1GWvSQmF1ZzZuW1uaTEi3fiJXgkWMEFXmKF4vU) (scroll down to the buttom)
image

File: apps/maestro/src/features/suiHooks/useDeployToken.ts [hunk 3]
```diff
@@ -135,14 +136,16 @@ export default function useTokenDeploy() {
         metadataId: metadata.objectId,
         destinationChains: destinationChainIds,
       });
-      // const sendTokenTx = Transaction.from(sendTokenTxJSON as string);
       const sendTokenResult = await signAndExecuteTransaction({
         transaction: sendTokenTxJSON as string,
         chain: "sui:testnet",
       });
+      const txIndex = sendTokenResult?.events?.[0]?.id?.eventSeq ?? 0;
```
Comment (npty) [inline apps/maestro/src/features/suiHooks/useDeployToken.ts:143]: you don't need to loop over events to find the `ContratCall` event here?

**Pull request merged**